controllers/gitrepository_controller.go

```diff
@@ -459,27 +482,77 @@ func (r *GitRepositoryReconciler) reconcileSource(ctx context.Context,
 		}
 	}
 
-	// Checkout HEAD of reference in object
-	gitCtx, cancel := context.WithTimeout(ctx, obj.Spec.Timeout.Duration)
-	defer cancel()
-	c, err := checkoutStrategy.Checkout(gitCtx, dir, repositoryURL, authOpts)
+	// Fetch the included artifact metadata.
+	artifacts, err := r.fetchIncludes(ctx, obj)
 	if err != nil {
-		var v git.NoChangesError
-		if errors.As(err, &v) {
-			return sreconcile.ResultSuccess,
-				&serror.Waiting{Err: v, Reason: v.Message, RequeueAfter: obj.GetRequeueAfter()}
-		}
+		return sreconcile.ResultEmpty, err
+	}
 
-		e := &serror.Event{
-			Err:    fmt.Errorf("failed to checkout and determine revision: %w", err),
-			Reason: sourcev1.GitOperationFailedReason,
-		}
+	// Observe if the artifacts still match the previous included ones
+	if artifacts.Diff(obj.Status.IncludedArtifacts) {
```

Comment:
Maybe `.Differ`?

Reply:
Not a new API, would prefer to not change here.